cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -120,10 +123,12 @@ func (m *azureCache) regenerate() error {
 	if err != nil {
 		return err
 	}
-
+		
 	// Regenerate instance to node groups mapping.
 	newInstanceToNodeGroupCache := make(map[azureRef]cloudprovider.NodeGroup)
+	registeredNodeGroupIDs := sets.New[string]()
 	for _, ng := range m.registeredNodeGroups {
+		registeredNodeGroupIDs.Insert(ng.Id())
```

Comment:
This is either the nodepoolName or its vmss name. Need to spin up a cluster to check

Reply:
Looked on the cluster, this is the vmss.Name for vmss. Availablitysets its teh name of the availablity set. Going to ignore availability set implementation for this edge case and just proceed with vmss due to the time crunch.